tsdb/head_append.go

```diff
@@ -65,6 +66,17 @@ func (a *initAppender) AppendExemplar(ref storage.SeriesRef, l labels.Labels, e
 	return a.app.AppendExemplar(ref, l, e)
 }
 
+func (a *initAppender) AppendMetadata(ref storage.SeriesRef, m metadata.Metadata) (storage.SeriesRef, error) {
+	if a.app != nil {
+		return a.app.AppendMetadata(ref, m)
+	}
+
+	// TODO check if we'll be keeping track of time here as well.
+	// a.head.initTime(t)
```

Comment:
This is done for the other Appends, so I'd say it's probably expected to init the time here as well.